### ISSUE TITLE
Did you include available subgoals in the `gpt_system_message`, the prompt of the decision LLM?

### ISSUE THREAD

**Issue by sjlee-me** [PKU-RL/AdaRefiner#4]
TL;DR: "Should I include 'available subgoals' in the prompt for decision LLM to reproduce your experiment results?"

I'm reproducing your codes, but I find a weird point in the prompt for decision LLM, `gpt_system_message` in utils.py.

In your code, `gpt_system_message` includes the following texts.
"""
Available sub-goals are: <eat plant, attack zombie, attack skeleton, attack cow, chop tree, mine stone, mine coal, mine iron, mine diamond, drink water, chop grass, sleep, place stone, place crafting table, place furnace, place plant, make wood pickaxe, make stone pickaxe, make iron pickaxe, make wood sword, make stone sword, make iron sword, plant row, chop grass with wood pickaxe, vegetarianism, make workshop, survival, deforestation, work and sleep, gardening, wilderness survival>. You are asked to suggest 3 sub-goals from available sub-goals in one sentence without any other words.
"""

I found that, when the subgoals are included, Llama 3 70B and GPT-4 cannot decompose a goal into easy subgoals and only pick some subgoals in the provided available subgoals, even though I use the same example prompts in your paper.

However, when I exclude the available subgoals in the system prompt, I found that Llama3 70B can decompose the subgoal. Moreover, there is no 'available subgoals' in your paper decision prompts example.
So, I'm confused whether I should include the 'available subgoals' in the prompt or not.

Below codes are example.

When available subgoals are included, decision LLM cannot decompose subgoals.

```python
from vllm import LLM, SamplingParams
from transformers import AutoTokenizer

DECISION_LLM_PATH = 'neuralmagic/Meta-Llama-3-70B-Instruct-quantized.w4a16'
decision_sampling_params = SamplingParams(temperature=0.5, top_p=1.0, max_tokens=100)
decision_tokenizer = AutoTokenizer.from_pretrained(DECISION_LLM_PATH)
decision_tokenizer.pad_token = decision_tokenizer.eos_token

decision_model = LLM(model=DECISION_LLM_PATH,
                        tensor_parallel_size=4,
                        gpu_memory_utilization=0.5)

gpt_system_message = """You are a professional game analyst. A player is playing a 2D Minecraft game. You will also get analysis about this player from another analyst. Available sub-goals are: <eat plant, attack zombie, attack skeleton, attack cow, chop tree, mine stone, mine coal, mine iron, mine diamond, drink water, chop grass, sleep, place stone, place crafting table, place furnace, place plant, make wood pickaxe, make stone pickaxe, make iron pickaxe, make wood sword, make stone sword, make iron sword, plant row, chop grass with wood pickaxe, vegetarianism, make workshop, survival, deforestation, work and sleep, gardening, wilderness survival>. You are asked to suggest 3 sub-goals from available sub-goals in one sentence without any other words."""

gpt_user_message = """Player sees: <grass, water, cow>. Player status: <7 health, 5 food, 6 drink , 4 energy >. Past action: <sleep>. Past sub-goals: <eat cow, collect stone, place stone>. Analysis: <The player seems to struggle with understanding past sub-goals, possibly indicating an early stage in the learning process. To help the agent learn the skill of eating cow more quickly, please provide more detailed guidance.>"""

gpt_prompt = [
    {"role": "system", "content": gpt_system_message},
    {"role": "user", "content": gpt_user_message},
]

chat_prompts = decision_tokenizer.apply_chat_template(gpt_prompt, tokenize=False)

outputs = decision_model.generate(chat_prompts, decision_sampling_params)
generated_text = outputs[0].outputs[0].text
generated_text = generated_text.replace('<|start_header_id|>assistant<|end_header_id|>', '').strip()
generated_text
```

the result is `attack cow, chop grass, drink water`.

When available subgoals are not included, decision LLM can decompose subgoals.

```python
gpt_system_message = """You are a professional game analyst. A player is playing a 2D Minecraft game. You will also get analysis about this player from another analyst. You are asked to suggest 3 sub-goals from available sub-goals in one sentence without any other words."""

gpt_user_message = """Player sees: <grass, water, cow>. Player status: <7 health, 5 food, 6 drink , 4 energy >. Past action: <sleep>. Past sub-goals: <eat cow, collect stone, place stone>. Analysis: <The player seems to struggle with understanding past sub-goals, possibly indicating an early stage in the learning process. To help the agent learn the skill of eating cow more quickly, please provide more detailed guidance.>"""

gpt_prompt = [
    {"role": "system", "content": gpt_system_message},
    {"role": "user", "content": gpt_user_message},
]

chat_prompts = decision_tokenizer.apply_chat_template(gpt_prompt, tokenize=False)

outputs = decision_model.generate(chat_prompts, decision_sampling_params)
generated_text = outputs[0].outputs[0].text
generated_text = generated_text.replace('<|start_header_id|>assistant<|end_header_id|>', '').strip()
generated_text
```

the result is `Approach cow, attack cow, eat cow`.

this is the prompt example in the paper I tried to decompose a subgoal
<img width="506" alt="스크린샷 2024-07-17 오전 11 20 06" src="https://github.com/user-attachments/assets/81b96c63-2e00-4eec-a970-3e3ac0ef9a2d">

**Comment by zawnpn** [comment]
Hello, thank you for your interest in our work.

For some unknown reasons, the prompt in the code is an older version. Please follow the settings in our paper and do not include the *available subgoals*.

Additionally, just FYI, we used Llama2-7B and GPT-4/3.5 (0613 version api) in our paper. Using Llama3 and the latest versions of GPT may also result in potential output differences, so when using them, you may consider finding some adjustments for the prompts to ensure the desired output.
